hacking/target-clock-generation.md

```diff
@@ -0,0 +1,144 @@
+# Target Clock Generation
+
+The Black Magic Debug firmware generates a clock signal to the debug target for both JTAG and SWD
+by bitbanging, timed to the other signals involved in ths bus to generate transactions and exchange
+data back and forth. The bitbanging routines are not perfect at this and so generate a signal with
+an odd mark-to-space ratio and at different frequencies depending on which routine is in play.
+
+Due to of all of the above, this winds up requiring a very particular approach for measuring the
+clock frequency generated to produce a number to report to the user, to take clock speed requests
+and map them to delay loop iteration counts, internally called the *clock divider*, and represent them
+in the firmware.
+
+The aim of this page is to lay out the method used to derive these clock frequency numbers and the
+analytical tools used to then generate a calibration for a platform's `platform.h` header to make the
+reporting and interaction as usefully accurate as possible.
+
+This method comes with a [LibreOffice workbook](../_assets/target-clk-gen/BMP_Frequency_Division.ods)
+that acts as a tool for computing the offset and division factors for computing `target_clk_divider`
```

Comment:
```suggestion
that acts as a tool for calculating the offset and division factors for computing `target_clk_divider`
```
(repetition; see https://www.merriam-webster.com/dictionary/calculate and https://www.merriam-webster.com/dictionary/compute)